src/DeVinci_frontend/components/ChatBox.svelte

```diff
@@ -265,20 +262,28 @@
       </label>
       <input id="pdf_chat" type="file" accept=".pdf" on:change={uploadPdfToVectorDatabase} class="hidden text-sm text-gray-900 border border-gray-300 cursor-pointer bg-gray-50 ml-2">
       {#if loadingKnowledgeDatabase}
-        <p class="font-semibold text-gray-900 dark:text-gray-600">Loading your content into the local Knowledge Base for you...</p>
+        <p class="font-semibold text-gray-900">Loading your content into the local Knowledge Base for you...</p>
         <img class="h-12 mx-auto p-1 block" src={spinner} alt="loading animation" />
       {/if}
-      {#if !$chatModelIdInitiatedGlobal || messageGenerationInProgress}
+      {#if !$chatModelIdInitiatedGlobal}
         <input disabled type="text" id="chat" autofocus class="block mx-4 p-3 w-full text-sm text-gray-900 bg-white rounded-lg border border-gray-300 focus:ring-2 focus:outline-none focus:ring-[#24292F]/50 " />
         <button disabled type="submit" class="opacity-55 cursor-not-allowed inline-flex justify-center p-2 text-gray-600 rounded-full">
           <svg class="w-5 h-5 rotate-0 rtl:-rotate-90" aria-hidden="true" xmlns="http://www.w3.org/2000/svg" fill="currentColor" viewBox="0 0 18 20">
             <path d="m17.914 18.594-8-18a1 1 0 0 0-1.828 0l-8 18a1 1 0 0 0 1.157 1.376L8 18.281V9a1 1 0 0 1 2 0v9.281l6.758 1.689a1 1 0 0 0 1.156-1.376Z"/>
           </svg>
           <span class="sr-only">Send message</span>
         </button>
+      {:else if messageGenerationInProgress}
+        <input disabled type="text" id="chat" class="block mx-4 p-3 w-full text-sm text-gray-900 bg-white rounded-lg border border-gray-300 focus:ring-2 focus:outline-none focus:ring-[#24292F]/50 " />
+        <button type="submit" class="inline-flex justify-center p-2 text-gray-600 rounded-full bg-gray-100 hover:bg-gray-300">
```

Comment:
to implement the stop functionality, you can add the on click event similarly to the button below (on:click={() => {sendMessage()}), e.g. as
on:click={() => {interruptMessageGeneration()}

where the function uses the functionality above
await $chatModelGlobal.interruptGenerate();

Reply:
added to development